src/humanize.cr

```diff
@@ -152,17 +152,17 @@ struct Number
   # delimiter (see `#format`).
   #
   # See `Int#humanize_bytes` to format a file size.
-  def humanize(io : IO, precision = 3, separator = '.', delimiter = ',', *, base = 10 ** 3, significant = true, prefixes : Indexable = SI_PREFIXES) : Nil
-    humanize(io, precision, separator, delimiter, base: base, significant: significant) do |magnitude, _|
+  def humanize(io : IO, precision = 3, separator = '.', delimiter = ',', *, base = 10 ** 3, significant = true, unit_separator : String = "", prefixes : Indexable = SI_PREFIXES) : Nil
```

Comment:
Other parameters are not specifying the type restrictions, also there's no need to disallow `Char` or `nil` as a valid arguments.

```suggestion
  def humanize(io : IO, precision = 3, separator = '.', delimiter = ',', *, base = 10 ** 3, significant = true, unit_separator = nil, prefixes : Indexable = SI_PREFIXES) : Nil
```

ditto elsewhere

Reply:
Also, the order of positional arguments shouldn't be changed - at least not without a deprecation first.

Reply:
The parameters are not positional (note the `*` splat in the parameter list). So the order really doesn't matter for compatibility reasons. I think the current order is logical.

It's a good idea to drop the type restriction though. There's no reason to require `String`. `Char` would be fine as well. Or really any other value. We just need it to stringify.

Reply:
@straight-shoota Agree on removing the type restriction - will implement this change asap.